src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php

```diff
@@ -32,14 +32,39 @@ public function configureOptions(OptionsResolver $resolver): void
 
                 $choiceTranslationLocale = $options['choice_translation_locale'];
 
+                // When the `active` and `legal_tender` options are null, it is unnecessary no need to retrieve the currencies from the meta indice.
+                if (null !== ($active = $options['active']) && null !== ($legalTender = $options['legal_tender'])) {
+                    $filteredCurrencyNames = [];
+
+                    foreach (Currencies::getCurrencyCodes() as $code) {
+                        if (!Currencies::isValidInAnyCountry($code, $legalTender, $active, $options['date'])) {
+                            continue;
+                        }
+
+                        $filteredCurrencyNames[$code] = Currencies::getName($code, $choiceTranslationLocale);
+                    }
+
+                    return ChoiceList::loader(
+                        $this,
+                        new IntlCallbackChoiceLoader(static fn () => array_flip($filteredCurrencyNames)),
+                        $choiceTranslationLocale
+                    );
+                }
+
                 return ChoiceList::loader($this, new IntlCallbackChoiceLoader(static fn () => array_flip(Currencies::getNames($choiceTranslationLocale))), $choiceTranslationLocale);
             },
             'choice_translation_domain' => false,
             'choice_translation_locale' => null,
+            'active' => null,
+            'legal_tender' => null,
+            'date' => new \DateTimeImmutable('today', new \DateTimeZone('Etc/UTC')),
```

Comment:
I'd go with filtering by default - there's no BC to preserve IMHO (but let's add a mention in the upgrade and changelog files):
```suggestion
            'valid_at' => new \DateTimeImmutable(),
            'legal_tender' => true,
```

Reply:
There may be a BC for applications that were still working with historical currencies (e.g. a museum ?), as the same results will not be returned by default.

However, this is acceptable if it is clearly indicated as you said.

Reply:
Moreover, if I replace the `date` and `active` options with a single `valid_at` datetime, it will no longer be possible to filter currencies that are *not* active on the given date.

For example, if your app needs to search invoices that were issued with the wrong currencies and show a dropdown of those invalid currencies, you would still need two parameters (`active` and `date`).